examples/spring-boot-with-certmanager-example/README.md

```diff
@@ -0,0 +1,122 @@
+
+```
+dekorate.options.input-path=k8s
+```
+
+Let's now configure the certificate to generate the previous Keystore:
+
+```
+dekorate.certificate.secret-name=tls-secret
+dekorate.certificate.self-signed.enabled=true
+dekorate.certificate.keystores.pkcs12.create=true
+# the secret name of the password (see above):
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.name=pkcs12-pass 
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.key=password
+```
+
+Using this configuration, Cert-Manager will generate a self-signed certificate and the Keystore files within the secret `tls-secret`. Therefore, we need to map the data within this secret `tls-secret` into the Spring Boot properties `server.ssl.key-store` and `server.ssl.key-store-password`.
+
+Moreover, the Cert-Manager Dekorate extension will configure the application resources to mount a volume with the generated certificates at the path `/etc/certs` (it can be configured using `dekorate.certificate.volume-mount-path`).
+
+So, the final dekorate properties that we need to add are the following:
+
+```
+dekorate.kubernetes.env-vars[0].name=SERVER_SSL_KEY_STORE
+dekorate.kubernetes.env-vars[0].value=/etc/certs/keystore.p12
+dekorate.kubernetes.env-vars[1].name=SERVER_SSL_KEY_STORE_PASSWORD
+dekorate.kubernetes.env-vars[1].secret=pkcs12-pass
+dekorate.kubernetes.env-vars[1].value=password
+```
+
+## Run the application in Kubernetes
+
+First, make sure you have connected/logged to a kubernetes cluster and installed [Cert-Manager](https://cert-manager.io/docs/installation/) in the cluster.
+
+Next, we need to generate the manifests and push the application container image to our container registry:
+
+```
+mvn clean install -Ddekorate.push=true -Ddekorate.docker.registry=<your container registry. Example: quay.io> -Ddekorate.docker.group=<the container group. Example: user>
+```
+
+After running the above command, we will have the generated manifests in the file `target/classes/META-INF/dekorate/kubernetes.yml`. So, let's install it:
+
+```
+kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml
+```
+
+Now, we should see the secret resource named `tls-secret` that `Cert-Manager` created with the expected keystore file:
+
+```
+kubectl get secret/tls-secret -o yaml | grep keystore.p12
+```
+
+After some minutes, our application should be up and running:
+
+```
+kubectl get pods
+```
+
+Output:
+
+```
+NAME                                                    READY   STATUS    RESTARTS   AGE
+spring-boot-with-certmanager-example-566546987c-nj94n   1/1     Running   0          2m23s
+```
+
+Let's try it out by port-forwarding the port 8443:
+
+```
+kubectl port-forward spring-boot-with-certmanager-example-566546987c-nj94n 8443:8443
+```
+
+Then, if we browse to `https://localhost:8443/`, we should see `Hello world from HTTPS!`!
```

Comment:
Can you add also a curl request to show what the content of the generated certificate will look like (or maybe using openssl command) and refers to the previous properties such as

```
dekorate.certificate.common-name=dekorate-example
dekorate.certificate.self-signed.enabled=true
dekorate.certificate.subject.organizations=Dekorate,Community
dekorate.certificate.duration=2160h0m0s
dekorate.certificate.renewBefore=360h0m0s
dekorate.certificate.privateKey.algorithm=RSA
dekorate.certificate.privateKey.encoding=PKCS8
dekorate.certificate.privateKey.size=2048
```

Why ? That will help the users to understand what they should do to generate their own TLS Server certificate

Reply:
I think that showing what the content of a certificate should look like is out of scope for this guide.